pkg/ccl/importccl/read_import_pgdump.go

```
@@ -742,60 +757,79 @@ func (m *pgDumpReader) readFile(
+						continue
+					}
+					return errors.Errorf("unsupported setval %T arg: %v", fn.Exprs[0], fn.Exprs[0])
+				}
+				seqval, ok := fn.Exprs[1].(*tree.NumVal)
 				if !ok {
+					return errors.Errorf("unsupported setval %T arg: %v", fn.Exprs[1], fn.Exprs[1])
+				}
+				val, err := seqval.AsInt64()
+				if err != nil {
+					return errors.Wrap(err, "unsupported setval arg")
+				}
+				isCalled := false
+				if len(fn.Exprs) == 3 {
+					called, ok := fn.Exprs[2].(*tree.DBool)
+					if !ok {
+						return errors.Errorf("unsupported setval %T arg: %v", fn.Exprs[2], fn.Exprs[2])
+					}
+					isCalled = bool(*called)
+				}
+				name, err := parser.ParseTableName(seqname.RawString())
+				if err != nil {
 					break
 				}
-				isCalled = bool(*called)
-			}
-			name, err := parser.ParseTableName(seqname.RawString())
-			if err != nil {
-				break
-			}
-			seq := m.descs[name.Parts[0]]
-			if seq == nil {
-				break
-			}
-			key, val, err := sql.MakeSequenceKeyVal(keys.TODOSQLCodec, seq.Desc, val, isCalled)
-			if err != nil {
-				return wrapRowErr(err, "", count, pgcode.Uncategorized, "")
-			}
-			kv := roachpb.KeyValue{Key: key}
-			kv.Value.SetInt(val)
-			m.kvCh <- row.KVBatch{
-				Source: inputIdx, KVs: []roachpb.KeyValue{kv}, Progress: input.ReadFraction(),
+				seq := m.descs[name.Parts[0]]
+				if seq == nil {
+					break
+				}
+				key, val, err := sql.MakeSequenceKeyVal(keys.TODOSQLCodec, seq.Desc, val, isCalled)
+				if err != nil {
+					return wrapRowErr(err, "", count, pgcode.Uncategorized, "")
+				}
+				kv := roachpb.KeyValue{Key: key}
+				kv.Value.SetInt(val)
+				m.kvCh <- row.KVBatch{
+					Source: inputIdx, KVs: []roachpb.KeyValue{kv}, Progress: input.ReadFraction(),
+				}
+			case "addgeometrycolumn":
+				// handled during schema extraction.
+			default:
+				return errors.Errorf("unsupported function: %s", funcName)
 			}
+		case *tree.SetVar, *tree.BeginTransaction, *tree.CommitTransaction:
+			// ignored.
```

Is it worth making more noise on some of the things that we're silently ignoring? Possibly a warning log? E.g. if users expect `SETs` to be applied?

---

I'm not sure. If we were ignoring other things I'd be more inclined to but we expect the dumpfile to contain these and know they have no effect (they'd only mutate the session vars, but we're not in a session -- we're just importing the data).